client/profile-wizard/header.js

```diff
@@ -71,29 +72,30 @@ export default class ProfileWizardHeader extends Component {
 			( step ) => step.key === currentStep
 		);
 
-		visibleSteps.map( ( step, index ) => {
+		visibleSteps.forEach( ( step, index ) => {
 			const previousStep = visibleSteps[ index - 1 ];
+			step.isComplete = step.isComplete || index < currentStepIndex;
 
-			if ( index < currentStepIndex ) {
-				step.isComplete = true;
-			}
+			const canClickStepLabel =
+				! previousStep || previousStep.isComplete || step.isComplete;
```

Comment:
Previously we used the `previousStep.isComplete` to decide whether the tab is clickable or not. This made the industry tab not clickable after going back.